### ISSUE TITLE
Unabled to use provided adaptors

### ISSUE THREAD

**Issue by LarBearrr** [okta/okta-jwt-verifier-php#33]
I have followed the setup instructions and have a basic PHP backend to verify the JWT token issued from one of our mobile applications. Neither the SpomkyLabsJose provider or Firebase provider are working:
```
$jwtVerifier = (new \Okta\JwtVerifier\JwtVerifierBuilder())
    ->setDiscovery(new \Okta\JwtVerifier\Discovery\Oauth) // This is not needed if using oauth.  The other option is OIDC
    ->setAdaptor(new \Okta\JwtVerifier\Adaptors\SpomkyLabsJose())
    ->setAudience('api://default')
    ->setClientId('{{CLIENT_ID}}')
    ->setIssuer('https://{{ORG__URL}}.com/oauth2/default')
    ->build();

$jwt = $jwtVerifier->verify($jwt);

dump($jwtVerifier);
```
is producing this error:

`filter_var(): explicit use of FILTER_FLAG_SCHEME_REQUIRED and FILTER_FLAG_HOST_REQUIRED is deprecated`

Switching to the Firebase adaptor produces:
`"kid" invalid, unable to lookup correct key`
which is mentioned in #17 

It seems the SpomkyLabsJose adaptor is deprecated as mentioned in #27 and there is missing information for the other adaptor. What can I do to resolve this as I need to validate JWTs issued from the mobile client. Any help is appreciated! Thank you!

**Comment by sleekuser** [comment]
Were you able to get any help on this issue?

**Comment by sleekuser** [comment]
@LarBearrr I was able to fix the issue eventually. The mobile issuer didn't match "exactly" with the URI in this library. For eg. In mobile application I had issuer as "http://{organization}.oktapreview.com" while was I using "http://{organization}.oktapreview.com/oauth2/default" in the PHP code. The issuer gets embedded in the JWT and that's what causing an issue for me.

**Comment by LarBearrr** [comment]
@sleekuser Yeah so it turned out to be similar. There were a couple of issues. First, we were using the {orgUrl} in the mobile app as the issuer instead of https://{orgUrl}/oauth2/default, which was causing the `kid` to not match. Once we figured that out, we started getting errors about the sub claim being invalid. Working with Okta support and a bit of research revealed that the username attribute on the user profile for the OIDC client needed to be set. After checking, it turned out the username was not set for the user. Once we set this value the kid matched and we were able to use the JWT verifier in our middleware.
